packages/@aws-cdk/aws-kinesisfirehose/lib/destination.ts

```diff
@@ -0,0 +1,99 @@
+import * as logs from '@aws-cdk/aws-logs';
+import { Construct } from 'constructs';
+import { IDeliveryStream } from './delivery-stream';
+import { CfnDeliveryStream } from './kinesisfirehose.generated';
+
+/**
+ * A Kinesis Data Firehose delivery stream destination configuration.
+ */
+export interface DestinationConfig {
+  /**
+   * Schema-less properties that will be injected directly into `CfnDeliveryStream`.
+   */
+  readonly properties: object;
+}
+
+/**
+ * Options when binding a destination to a delivery stream.
+ */
+export interface DestinationBindOptions {
+  /**
+   * The delivery stream.
+   */
+  readonly deliveryStream: IDeliveryStream;
+}
+
+/**
+ * A Kinesis Data Firehose delivery stream destination.
+ */
+export interface IDestination {
+  /**
+   * Binds this destination to the Kinesis Data Firehose delivery stream.
+   *
+   * Implementers should use this method to bind resources to the stack and initialize values using the provided stream.
+   */
+  bind(scope: Construct, options: DestinationBindOptions): DestinationConfig;
+}
+
+/**
+ * Generic properties for defining a delivery stream destination.
+ */
+export interface DestinationProps {
+  /**
+   * If true, log errors when data transformation or data delivery fails.
+   *
+   * If `logGroup` is provided, this will be implicitly set to `true`.
+   *
+   * @default true - errors are logged.
+   */
+  readonly logging?: boolean;
+
+  /**
+   * The CloudWatch log group where log streams will be created to hold error logs.
+   *
+   * @default - if `logging` is set to `true`, a log group will be created for you.
+   */
+  readonly logGroup?: logs.ILogGroup;
+}
+
+/**
+ * Abstract base class that destination types can extend to benefit from methods that create generic configuration.
+ */
+export abstract class DestinationBase implements IDestination {
+  private logGroups: { [logGroupId: string]: logs.ILogGroup } = {};
+
+  constructor(protected readonly props: DestinationProps = {}) { }
+
+  abstract bind(scope: Construct, options: DestinationBindOptions): DestinationConfig;
+
+  protected createLoggingOptions(
+    scope: Construct,
+    deliveryStream: IDeliveryStream,
+    streamId: string,
+  ): CfnDeliveryStream.CloudWatchLoggingOptionsProperty | undefined {
+    return this._createLoggingOptions(scope, deliveryStream, streamId, 'LogGroup', this.props.logging, this.props.logGroup);
+  }
+
+  private _createLoggingOptions(
+    scope: Construct,
+    deliveryStream: IDeliveryStream,
+    streamId: string,
+    logGroupId: string,
+    logging?: boolean,
+    propsLogGroup?: logs.ILogGroup,
+  ): CfnDeliveryStream.CloudWatchLoggingOptionsProperty | undefined {
+    if (logging === false && propsLogGroup) {
+      throw new Error('logging cannot be set to false when logGroup is provided');
+    }
+    if (logging !== false || propsLogGroup) {
+      this.logGroups[logGroupId] = this.logGroups[logGroupId] ?? propsLogGroup ?? new logs.LogGroup(scope, logGroupId);
```

Comment:
You can query children of a node at runtime with `scope.node.tryFindChild(logGroupId)`. I feel like this will be more robust than the caching in a local field (imagine what would happen if the same Destination was used for 2 different DeliveryStreams in different Stacks, for example).

Reply:
Oh thank you! I will try that, I thought there might be a way to leverage the construct here, but wasn't sure how exactly.  Also for context, the reason this is implemented this way is because in future changes there will also be a "BackupLogGroup" for logging for the backup destination.

Reply:
@skinny85 I might be missing something here, but `Construct` doesn't have a `node` attribute. Should I use `deliveryStream.node`? That doesn't seem quite right.

Reply:
Maybe it's a `constructs.Construct`? In that case, I believe the API is `constructs.Node.of(myConstruct)`.

Reply:
Oh yep, that was it. thank you!